### PR TITLE
Add api validation support.

### PR DESCRIPTION
**Pull request by Bonkles**
Was able to successfully get this to validate existing APIs using the api swagger validation service. 

Example of swagger that will generate api validation errors:  https://developer.cimpress.io/2016-04-01_11-33_swagger-api3.json

### CODE REVIEW

File: lib/cimpress_mcp/cli.rb [hunk 1]
```diff
@@ -81,12 +81,21 @@ def main(argv)
...
+			else
+				puts "Success. No errors found."
+			end
 		when 'get_fulfillment_recommendations'
 			puts mcp.get_fulfillment_recommendations(sku: 'VIP-44525', quantity: 250, country: 'US', postal_code: '01331')
 		when 'create_barcode'
 			createBarcodeResponse = mcp.create_barcode()
 			puts createBarcodeResponse
-			when 'health_check'
+    when 'health_check'
```
Comment (seonggbang) [inline lib/cimpress_mcp/cli.rb]: Can we match the indentation here?

File: lib/cimpress_mcp/client.rb [hunk 2]
```diff
@@ -162,6 +162,24 @@ def get_fulfillment_recommendations(sku:, quantity:, country:, postal_code:)
         return JSON.parse(response)
     end
 
+    def validate_api(swagger_url_to_validate:)
+        RestClient.log=STDOUT
+        validate_request = <<-DOC
+            {"swagger_url": "#{swagger_url_to_validate}"}
```
Comment (sroylance) [inline lib/cimpress_mcp/client.rb]: I realize I set the bad precedent,  but I think we should switch to using the JSON library to generate requests documents

File: lib/cimpress_mcp/cli.rb [hunk 3]
```diff
@@ -103,3 +112,4 @@ def main(argv)
 
 	end
 end
+end
```
Comment (seonggbang) [inline lib/cimpress_mcp/cli.rb]: I don't see any new opening statement to warrant this.

File: lib/cimpress_mcp/client.rb [hunk 4]
```diff
@@ -162,6 +162,24 @@ def get_fulfillment_recommendations(sku:, quantity:, country:, postal_code:)
         return JSON.parse(response)
     end
 
+    def validate_api(swagger_url_to_validate:)
+        RestClient.log=STDOUT
```
Comment (sroylance) [inline lib/cimpress_mcp/client.rb]: probably don't want this, at least by default, maybe hide it behind a verbose arg that default to false?

File: lib/cimpress_mcp/client.rb [hunk 5]
```diff
@@ -162,6 +162,24 @@ def get_fulfillment_recommendations(sku:, quantity:, country:, postal_code:)
         return JSON.parse(response)
     end
 
+    def validate_api(swagger_url_to_validate:)
```
Comment (seonggbang) [inline lib/cimpress_mcp/client.rb]: api_url should be sufficient enough unless you wanted to validate swagger documentation. If so, then you should name this to validate_swagger_documentation instead.

File: lib/cimpress_mcp/client.rb (hunk 2 above)
Comment (Bonkles) [inline lib/cimpress_mcp/client.rb]: Great idea.  I'll replace all uses of the DOC body generation with the json library generation.

File: lib/cimpress_mcp/cli.rb (hunk 1 above)
Comment (Bonkles) [inline lib/cimpress_mcp/cli.rb]: Donnnnne

File: lib/cimpress_mcp/cli.rb (hunk 3 above)
Comment (Bonkles) [inline lib/cimpress_mcp/cli.rb]: Ah, this actually closes the 'Module cimpress_mcp' directive in line one.

File: lib/cimpress_mcp/client.rb (hunk 5 above)
Comment (Bonkles) [inline lib/cimpress_mcp/client.rb]: Leaving as-is- we're taking the swagger url as an input, but we're not actually validating swagger syntax, we're just making sure that certain cimpress-wide conventions are being adhered to. We can figure out what to call this when it's more fully baked.

**Pull request merged**